### ISSUE TITLE
JAR dependencies on ubuntu

### ISSUE THREAD

**Issue by D3SL** [miraisolutions/xlconnect#139]
If I understood previous closed issues correctly (#138, #132, #122) these _are_ still necessary on linux, correct? I'm using XLConnect with Shinyproxy, meaning in docker containers. If these are still necessary for Ubuntu Focal Fossa is it possible to have these downloaded and installed along with other packages during the docker image build process? Currently they auto-download during application start which adds a ~5ish seconds to app start time.

Interestingly the package still gives a warning even though it appears from the logs that all the java dependencies did download successfully, is this a configuration error on my part perhaps?

```
XLConnect: It seems downloading the JAR dependencies may have failed. If you would like to use a different maven repository, please set the environment variable XLCONNECT_JAVA_REPO_URL to a valid URL, e.g. Sys.setenv(XLCONNECT_JAVA_REPO_URL='https://jcenter.bintray.com'), and reinstall the package.
```

**Comment by spoltier** [comment]
Hi @D3SL, are you using version 1.0.3 ?

If you install XLConnect during the docker image build, the jar dependencies should download at that time. 

I tried this with a Dockerfile based on _rocker/r-devel_, and I could see the dependencies being downloaded in the `docker build ...` output.
I used `RUN install2.r --error XLConnect` to install XLConnect. There is probably a different way to do it with Shinyproxy / the base image you're using.When running a container of the resulting image with `Rscript -e "library(XLConnect)"` I didn't see any failure or downloading of the packages.

The failure message could indicate that the build environment doesn't / didn't have access to the internet to download the dependencies. In `onLoad` we assume that the package is being installed if `interactive()` returns `FALSE`, but I suppose when running under Shinyproxy that might be the case as well. We only try downloading if the dependencies are missing however.

Re: the exact linux distro, this shouldn't make a difference, as we always have some jar which isn't available as deb/rpm/etc. in the official package repository, so we always have to download something.

**Comment by spoltier** [comment]
@D3SL we recently released 1.0.4, it would be nice if you tried again with that and let us know if anything changes (along with addressing my previous comment).

**Comment by D3SL** [comment]
Apologies for the delayed response, I have finally had the chance to try out the approach you mentioned on June 3rd and it seems that this takes care of it. The installation process is handled as part of the multi-stage image build and when the actual app is run it no longer needs to redo it every time.

Thank you for the help
